torch/_functorch/aot_autograd.py

```diff
@@ -2255,6 +2288,77 @@ def runtime_wrapper(*args):
             return fw_outs
     return runtime_wrapper
 
+@contextmanager
+def patch(module, attr, patch_fn):
+    """
+    Context manager that temporarily patches attribute of a module to a given patch_fn
+    """
+    old_fn = getattr(module, attr)
+    setattr(module, attr, patch_fn)
+    try:
+        yield
+    finally:
+        setattr(module, attr, old_fn)
+
+def create_functionalized_rng_ops_wrapper(func, args, trace_joint=True):
+    # Functionalization of rng ops changes the calling convention of the joint graph.
+    # It goes from (primals, tangents) to (seed, offset, primals, tangents)
+    # At runtime, we pass on the current seed and offset. This is hidden from
+    # the user.
+    fake_mode_init = False
+    for x in args:
+        if isinstance(x, FakeTensor):
+            fake_mode = x.fake_mode
+            fake_mode_init = True
+            break
+
+    if not fake_mode_init:
```

Comment:
Didn't Ed say that fake mode should be providing such API to get or initialize fake mode this hasn't happened yet?